### PR TITLE
PLAT-11401: Support variables usage for list/map attributes

### PR DESCRIPTION
**Pull request by symphony-youri**
In activity attributes with type list or map it was not possible to use
variable references (${}).

How are variables handled in workflows?

In SWADL you use them with ${}
When SWADL is deserialized to the Java model, strings were used
The Java model activity is serialized to JSON and stored in BPMN
When the BPMN is executed, expression (including variables) are
resolved and real values are replaced.
When the CamundaExecutor runs, the JSON activity is deserialised and the
Java mode rebuilt to be used in the Activity executor.

We now use a custom Jackson deserializer to be able to store in the Java
model either a resolved value or a variable reference (i.e the Variable
class).

We then use a custom TypeConverter during expression resolution (juel)
in Camunda to store complex objects (list,map) as escaped JSON.

When we deserialize to rebuild the Java model in CamundaExecutor another
custom deserializer is used to parse this escaped JSON and preserve the
complex objects.

Activities that have attributes other than strings should wrap them in
Variable<?>. Given JSON only support a generic number type, it is
difficult to differentiate a float/integer/long so it is up to the
activity developer to handle it.

The schema has been changed to allow to use variables or lists or maps
for the attributes that support it.

### CODE REVIEW

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/camunda/variable/EscapedJsonVariableDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import java.util.Map;
+
+
+/**
+ * Variables are serialized as escaped JSON by {@link VariableToJsonConverter} and passed to the executor.
+ * When the executor deserializes the activity to access the model, it uses this custom deserializer to
+ * recreate the {@link Variable} objects now containing resolved values.
+ */
+@SuppressWarnings("rawtypes")
+public class EscapedJsonVariableDeserializer extends StdDeserializer<Variable> {
```
Comment (symphony-soufiane) [inline workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/camunda/variable/EscapedJsonVariableDeserializer.java:26]: Shouldn't this be moved to workflow/swadl/ like ActivityDeserializer?

Comment (symphony-youri) [reply workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/camunda/variable/EscapedJsonVariableDeserializer.java:26]: this used by CamundaExecutor and relate to how expressions are resolved (so specific to Camunda)

File: workflow-language/src/main/java/com/symphony/bdk/workflow/swadl/v1/Variable.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    this.resolvedValue = resolvedValue;
+  }
+
+  @JsonIgnore
+  public T get() {
+    return resolvedValue;
+  }
+
+  @JsonIgnore
+  public Integer getInt() {
```
Comment (symphony-soufiane) [inline workflow-language/src/main/java/com/symphony/bdk/workflow/swadl/v1/Variable.java:41]: why only getInt() ? are the other types all handled by the method above?

**Review by symphony-soufiane — APPROVED** [review]

File: workflow-language/src/main/java/com/symphony/bdk/workflow/swadl/v1/Variable.java (hunk 2 above)
Comment (symphony-youri) [reply workflow-language/src/main/java/com/symphony/bdk/workflow/swadl/v1/Variable.java:41]: It was the only one that i needed :) (but we could do the same for other types)
we use it for pagination/skip/limit
for user ids, it is often a list of long so the logic is a bit different and kept in the executors

**Review by symphony-elias — APPROVED** [review]

**Pull request merged**